### PR TITLE
Junos sanity fixes

### PR DESCRIPTION
**Pull request by rohitthakur2590**
Signed-off-by: Rohit Thakur <rohitthakur2590@outlook.com>

##### SUMMARY

resolves: https://github.com/ansible-collections/junipernetworks.junos/issues/77

##### ISSUE TYPE

- Bugfix Pull Request
 

##### COMPONENT NAME

junos_*
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/junos_interface.py [hunk 1]
```diff
@@ -255,12 +348,12 @@ def main():
     neighbors_spec = dict(host=dict(), port=dict())
 
     element_spec = dict(
-        name=dict(),
+        name=dict(required=True),
```
Comment (NilashishC) [inline plugins/modules/junos_interface.py:351]: top-level `name` cannot be required for modules having `aggregate` key

File: plugins/modules/junos_linkagg.py [hunk 2]
```diff
@@ -311,7 +364,7 @@ def main():
     element_spec = dict(
         name=dict(),
         mode=dict(default="on", choices=["on", "off", "active", "passive"]),
-        members=dict(type="list"),
+        members=dict(required=True, type="list", elements="str"),
```
Comment (NilashishC) [inline plugins/modules/junos_linkagg.py:367]: top-level `members` cannot be required

File: plugins/modules/junos_logging.py [hunk 3]
```diff
@@ -198,7 +262,6 @@ def main():
         rotate_frequency=dict(type="int"),
         size=dict(type="int"),
         files=dict(type="int"),
-        src_addr=dict(),
```
Comment (NilashishC) [inline plugins/modules/junos_logging.py:201]: Looks like an existing `key` is removed?

File: plugins/modules/junos_package.py [hunk 4]
```diff
@@ -174,7 +176,6 @@ def main():
         no_copy=dict(default=False, type="bool"),
         validate=dict(default=True, type="bool"),
         force=dict(type="bool", default=False),
-        transport=dict(default="netconf", choices=["netconf"]),
```
Comment (NilashishC) [inline plugins/modules/junos_package.py:177]: Same as above

File: plugins/modules/junos_scp.py [hunk 5]
```diff
@@ -126,13 +129,12 @@ def main():
...
     """
     argument_spec = dict(
-        src=dict(type="list", required=True),
+        src=dict(type="list", required=True, elements="str"),
         dest=dict(type="path", required=False, default="."),
         recursive=dict(type="bool", default=False),
         remote_src=dict(type="bool", default=False),
         ssh_private_key_file=dict(type="path"),
         ssh_config=dict(type="path"),
-        transport=dict(default="netconf", choices=["netconf"]),
```
Comment (NilashishC) [inline plugins/modules/junos_scp.py:135]: Same as above,

File: plugins/modules/junos_static_route.py [hunk 6]
```diff
@@ -168,8 +213,8 @@ def main():
     """ main entry point for module execution
     """
     element_spec = dict(
-        address=dict(aliases=["prefix"]),
-        next_hop=dict(),
+        address=dict(aliases=["prefix"], required=True),
```
Comment (NilashishC) [inline plugins/modules/junos_static_route.py:216]: Top level keys cannot be `required` for modules having `aggregate` option

File: plugins/modules/junos_vlan.py [hunk 7]
```diff
@@ -172,9 +225,9 @@ def main():
     """
     element_spec = dict(
         name=dict(),
-        vlan_id=dict(type="int"),
+        vlan_id=dict(type="int", required=True),
```
Comment (NilashishC) [inline plugins/modules/junos_vlan.py:228]: Top level keys cannot be required for modules having `aggregate` option

File: plugins/modules/junos_interface.py (hunk 1 above)
Comment (rohitthakur2590) [reply plugins/modules/junos_interface.py:351]: that's right!

File: plugins/modules/junos_logging.py (hunk 3 above)
Comment (rohitthakur2590) [reply plugins/modules/junos_logging.py:201]: yes, It was dead key.

File: plugins/modules/junos_package.py (hunk 4 above)
Comment (rohitthakur2590) [reply plugins/modules/junos_package.py:177]: This `transport` option not used anywhere.

File: plugins/modules/junos_static_route.py (hunk 6 above)
Comment (rohitthakur2590) [reply plugins/modules/junos_static_route.py:216]: Done

File: plugins/modules/junos_vlan.py (hunk 7 above)
Comment (rohitthakur2590) [reply plugins/modules/junos_vlan.py:228]: Done

**Comment by rohitthakur2590** [comment]
recheck

**Comment by rohitthakur2590** [comment]
recheck

**Pull request merged**